0000-deb-repo-sync-gpg-check.md

```diff
@@ -0,0 +1,178 @@
+
+### Repo metadata location and signatures
+
+_Note: this section is only informative_
+
+Debian/Ubuntu repos use GPG signing to ensure the integrity of the `Release` file. The signature can be either in a separate file `Release.gpg` or inline in a file called `InRelease`. If both files are present the `InRelease` file is preferred.
+
+Typically the structure of a repo is like this:
+```
+http://mirror.example.com/ubuntu  <- $ARCHIVE_ROOT
+|
++- dists/bionic                   <- Suite or Codename
+   |
+   +- main                        <- $COMPNENT directories
+      |
+      +- binary-amd64
+         |
+         +- Packages.gz           <- Packages indices
+         +- Packages.xz
+         . . .
+      +- binary-i386
+      . . .
+   +- universe
+   . . .
+   +- Release                     <- Plain file
+   +- InRelease                   <- Release file + inline signature
+   +- Release.gpg                 <- Separate signature
+
+```
+
+E.g. for official Ubuntu repos the URL of these files:
+```
+http://mirror.example.com/ubuntu/dists/bionic-updates/InRelease
+http://mirror.example.com/ubuntu/dists/bionic-updates/Release.gpg
+```
+
+There's also an alternative layout for Debian repos called "flat repos". In this case the repo directory containins both `Release*` and `Packages*` files. E.g.:
+```
+https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/Ubuntu1804-Uyuni-Client-Tools/xUbuntu_18.04/
+```
+
+_Note #1:_
+
+In Uyuni the URL of a deb repository must point to the directory containing the binary packages for a specific architecture. E.g. for Ubuntu 18.04 updates repo is something like:
+```
+http://mirror.example.com/ubuntu/dists/bionic-updates/main/binary-amd64/
+```
+or in case of a flat repository to the directory containing both `Release*` and `Packages*` files:
+```
+https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/Ubuntu1804-Uyuni-Client-Tools/xUbuntu_18.04/
+```
+
+_Note #2:_
+
+Debian repos don't have a standard location for the public GPG keys used to verify the metadata. Sometimes the key is located in the root of a flat repo but the location is not standardized.
+
+### GPG keys management
+
+Uyuni ships the SUSE and OpenSUSE keys in the package `uyuni-build-keys`. The keys for Ubuntu and Debian should be shipped in this package too.
+
+Any additional keys that need to be used (e.g. for Ubuntu derivatives distros or for custom repos) must be created manually using the existing UI (`Systems -> Autoinstallation -> GPG and SSL keys`)
+
+
+### GPG Key usage in the UI
+
+The UI for creating/updating a repository (`Software -> Manage -> Repositories`) must be extended with one additional field:
+- GPG key
+
+This field must be visible only if the selected repository type is `deb` and the checkbox `Has Signed Metadata` is checked. For `deb` repositories the `Has Signed Metadata` checkbox must be enabled by default.
```

Comment:
This suggest now a different way compared to the way a section before.
And it does it only for debian. Why not for RPM as well?

But this is new UI and different key handling as we have it now.
Do we really need to go that deep for a first step? Or is this already step 2?

Reply:
As a first step we could manage with shipping Ubuntu and Debian official keys in packages. I guess that's the main use case anyways and leave custom repos for a second phase.